### DISCUSSION TITLE
socket 关于异常错误  没有打印的问题

### DISCUSSION THREAD

**Discussion by lualcs** [cloudwu/skynet#1483] [General]
如果是在 socket.accept(fd,handle,"ws",addr)  之后因为自己 handle模块的问题导致异常错误，
我希望是会抛出一个错误日志，然后断开连接
但是这个时候是直接会断开连接。  这个时候我如果在自己的模块出错前加一个print 打印 就会抛出错误。

感觉是上一个动作是socket 的动作 然后中间没有打印直接然后遇到异常 这个时候会导致没有异常日志

**Comment by cloudwu** [comment]
需要测试代码

**Comment by lualcs** [comment]
我修改simplewebsocket 重现了这个问题&nbsp; &nbsp;代码我放在附件里面了（我用的是ubuntu）


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cloudwu/skynet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月7日(星期四) 中午12:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cloudwu/skynet] socket 关于异常错误  没有打印的问题 (Discussion #1483)





 
需要测试代码
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by cloudwu** [comment]
没有附件，你可以尝试给一个解决方案。

**Comment by lualcs** [comment]
local&nbsp;function&nbsp;try_handle(self,&nbsp;method,&nbsp;...)
&nbsp;&nbsp;&nbsp;&nbsp;local&nbsp;handle&nbsp;=&nbsp;self.handle
&nbsp;&nbsp;&nbsp;&nbsp;local&nbsp;f&nbsp;=&nbsp;handle&nbsp;and&nbsp;handle[method]
&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;f&nbsp;then
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;---加上pcall&nbsp;是因为&nbsp;lssues#1482
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;local&nbsp;ok,error&nbsp;=&nbsp;pcall(f,self.id,&nbsp;...)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;assert(not&nbsp;ok,error)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--f(self.id,&nbsp;...)
&nbsp;&nbsp;&nbsp;&nbsp;end
end


from lualib/http/websocket.lua 我将这里改了一下，有错误输出了 我不懂为什么 f(self.id,&nbsp;...) 调用发生错误会导致不输出错误 (所以没有根本的解决问题)





------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年10月7日(星期四) 晚上6:48
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [cloudwu/skynet] socket 关于异常错误  没有打印的问题 (Discussion #1483)





 
没有附件，你可以尝试给一个解决方案。
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by lualcs** [comment]
是&nbsp;lualib/http/websocket.lua&nbsp; line:423&nbsp;error(err)  这里句代码屏蔽掉了 （我不知道为什么要屏蔽掉-问题确认了）



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cloudwu/skynet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月7日(星期四) 晚上6:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cloudwu/skynet] socket 关于异常错误  没有打印的问题 (Discussion #1483)





 
没有附件，你可以尝试给一个解决方案。
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by cloudwu** [comment]
如果你不会用 email , 请在 web 界面编辑清楚。

**Comment by lualcs** [comment]
不好意思是因为  lualib/http/websocket.lua  
里面 function M.accept(socket_id, handle, protocol, addr, options)
使用了xpcall
 error(err)  这句代码注释掉了所以没有错误输出  不明白为什么要注释这句代码

**Comment by cloudwu** [comment]
因为这个函数有返回 false, err 。你应该检查其返回值。
